src/features/projects/projectsSlice.js

```diff
@@ -109,6 +119,37 @@ export const projectsSlice = createSlice({
       state.unsavedViewChanges = payload;
     },
 
+    createProjectStart: (state) => {
+      const ls = { isLoading: true, operation: 'fetching', errors: null, stateMsg: null };
+      state.loadingStates.createProject = ls;
+    },
+
+    createProjectSuccess: (state, { payload }) => {
+      const { project } = payload.createProject;
+      const ls = {
+        isLoading: false,
+        operation: null,
+        errors: null,
+        stateMsg: `Successfully created project ${project.name}`
+      };
+      state.loadingStates.createProject = ls;
+
+      state.projects = {
```

Comment:
After I successfully create a project and try to navigate to the app by clicking the "application" nav link, the app crashes, and it looks like it may be because there's a bug here and all of the previous `projects` in state get accidentally overwritten with the newly created project. Your code here looks like it should work to me so I'm not totally sure what's going on...

Reply:
There was a bug, `projects` should be an array not an object. I fixed that but now I'm seeing another error `TypeError: Cannot read property 'roles' of undefined` but I'm not sure what this is related to.

Reply:
Hmm, I'm not seeing that new error.

However, I'm noticing another related bug. It's a little odd and I'm not sure how consequential so I'm not sure how much time to dedicate towards a fix, but the gist is:

- for a super user to access the animl.camera/create-project form, they would either have to be signed in already or navigate to animl.camera/app and sign in to get their JWT with their creds. After being authenticated, the `ProjectAndViewNav.jsx` component looks to see if there are any projects in state, and if not, it fetches them. So at this point the user would presumably have some projects already in state.
- if they they want to create a project, they'd have to manually type out that URL because there's no link to it, and when they navigate there it's a page refresh the projects get removed from state.
- once the user successfully creates a project, that gets added to the state, BUT if they then navigate back to the app using the "application" nav link, the state is preserved and the `ProjectAndViewNav.jsx` component sees that there are already projects in the state so it doesn't re-fetch them.

There are probably some fairly straightforward ways to fix this (maybe instead of housing the create-projects form on a separate page, we put it in a modal), but I am not overly concerned about this so I think let's call it good enough for now.